std/algorithm/mutation.d

```diff
@@ -2389,6 +2374,49 @@ void swap(T)(ref T lhs, ref T rhs) if (is(typeof(lhs.proxySwap(rhs))))
     lhs.proxySwap(rhs);
 }
 
+/**
+Swaps two elements in-place of a random access range `r`,
+specified by their indices `i1` and `i2`.
+
+Params:
+    r  = a $(XREF_PACK_NAMED range,primitives,isRandomAccess,random access range)
+        with swappable elements
+    i1 = first index
+    i2 = second index
+*/
+void swapAt(R)(R r, size_t i1, size_t i2)
```

Comment:
Studying this more closely... some types that should work as `R`, but which do not meet the formal definition of a "*random access range*":

- `char[]` and `wchar[]`, because of auto-decoding
- Static arrays
- Custom types that implement `opIndex()` to return an *lvalue*
- Custom types that have a `moveAt()` method *and* implement `opIndex()` with a return type that overloads `opAssign()` or has an assignable `alias _ this` - even if `R`'s `opIndex()` returns an *rvalue*

So, we can either write an elaborate explanation of the true requirements, or just change `a random access range` to `a range` and let people guess-and-check / read the source code.